sql/database.go

```diff
@@ -292,16 +330,194 @@ func Open(uri string, opts ...Opt) (*sqliteDatabase, error) {
 	return db, nil
 }
 
+func ensureDBSchemaUpToDate(logger *zap.Logger, db *sqliteDatabase, config *conf) (*sqliteDatabase, error) {
+	before, after, err := config.schema.CheckDBVersion(logger, db)
+	switch {
+	case err != nil:
+		return db, err
```

Comment:
better to instrument the error before returning it

Reply:
Fixed